### PR TITLE
Switch to using Sponge Task instead of our own Thread

### PR DESCRIPTION
**Pull request by parlough**
This is more of a personal preference, as I feel like it is much cleaner and it will be easier to maintain as time goes on. With my limited knowledge of Java threading, this seems to have the same functionality.

### CODE REVIEW

**Comment by parlough** [comment]
We could easily use a lambda and get rid of the `RecordingQueueManager` but I feel it helps with organization to keep it out of the main class.

File: src/main/java/com/helion3/prism/Prism.java [hunk 1]
```diff
@@ -145,7 +147,12 @@ else if (engine.equalsIgnoreCase("mysql")) {
         }
 
         // Initialize the recording queue manager
-        new RecordingQueueManager().start();
+        Task.builder()
+            .async()
+            .name("PrismRecordingQueueManager")
+            .interval(1000, TimeUnit.MILLISECONDS)
```
Comment (viveleroi) [inline src/main/java/com/helion3/prism/Prism.java]: Is there a `TimeUnit.SECONDS`? I would imagine that would be cleaner:

`.interval(1, TimeUnit.SECONDS)`

vs

`.interval(1000, TimeUnit.MILLISECONDS)`

Comment (parlough) [reply src/main/java/com/helion3/prism/Prism.java]: Yeah there is, I'll make that change :)

**Comment by viveleroi** [comment]
Looks good, have you tested this? I assume you have, but don't want to assume. I would have used this, Bukkit had something similar, but I'm pretty sure the scheduler stuff didn't exist in Sponge when I first wrote this.

**Comment by parlough** [comment]
@viveleroi Yeah I have tested this(with H2 backend but it shouldn't make a difference) and functionality appears to be exactly the same as before. Is the name of the task good or would you suggest something different?

Oh, and I switched it to 1 second.

**Comment by viveleroi** [comment]
The name is good with me.

**Pull request merged**